themes/bootstrap3/templates/search/facet-list.phtml

```diff
@@ -28,36 +26,45 @@
     <label><?=$this->translate('Sort') ?></label>
     <div class="btn-group">
       <?php foreach ($this->sortOptions as $key => $sort): ?>
-        <a href="<?=$urlBase . '&amp;facetpage=1&amp;facetsort=' . urlencode($key) ?>" class="btn btn-default js-facet-sort<?php if ($this->sort == $key): ?> active<?php endif; ?>" data-sort="<?=$key ?>" data-lightbox-ignore><?=$this->translate($sort) ?></a>
+        <a href="<?=$this->escapeHtmlAttr($urlBase . '&facetpage=1&facetsort=' . urlencode($key) . '&contains=' . urlencode($contains)) ?>" class="btn btn-default js-facet-sort<?php if($this->sort == $key): ?> active<?php endif; ?>" data-sort="<?=$key ?>"><?=$this->translate($sort) ?></a>
       <?php endforeach; ?>
     </div>
+
+    <?php if (method_exists($this->results->getParams(), 'setFacetContains')): ?>
+      <label><?=$this->transEsc('Filter') ?></label>
+      <div class="btn-group">
+        <form>
+          <input class="ajax_param" data-name="contains" class="form-control" type="text" value="<?=htmlspecialchars($this->contains)?>" aria-label="<?=$this->transEscAttr('search_terms')?>">
+          <button id="btn-reset-contains" type="reset" role="button" aria-label="<?=$this->transEsc('searchform_reset_button')?>"><?=$this->icon('trash')?></button>
```

Comment:
I wonder if it would be better to create a new `clear_facet_filter` language string by copying the existing `clear_feedback_filter` or `clear_tag_filter` translations to make this easier to customize/differentiate as needed. If you'd rather keep this as-is, though, that's okay with me at least as a first step; we can always revise in the next release if there's a strong need to clarify things.

Reply:
As you already mentioned, i also think it's better to leave it as-is for now and introduce a seperate language string later if there is a strong need for it.

Reply:
That makes sense, but I think in that case, it may increase the importance of my proposed fieldset/legend solution to associate the form fields with a specific context, so a screen reader can more easily tell the difference between the facet filter control and the search box. If all text is identical, it will otherwise be difficult to understand the context.